src/Symfony/Component/Serializer/Command/DebugCommand.php

```diff
@@ -0,0 +1,168 @@
+The <info>%command.name% 'App\Entity\Dummy'</info> command dumps the serializer groups for the dummy class.
+
+The <info>%command.name% src/</info> command dumps the serializer information for the `src` directory.
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $class = $input->getArgument('class');
+
+        if (class_exists($class)) {
+            $this->dumpSerializerDataForClass($input, $output, $class);
+
+            return 0;
+        }
+
+        try {
+            foreach ($this->getResourcesByPath($class) as $class) {
+                $this->dumpSerializerDataForClass($input, $output, $class);
+            }
+        } catch (DirectoryNotFoundException) {
+            $io = new SymfonyStyle($input, $output);
+            $io->error(sprintf('Neither class nor path were found with "%s" argument.', $input->getArgument('class')));
+
+            return 1;
+        }
+
+        return 0;
+    }
+
+    private function dumpSerializerDataForClass(InputInterface $input, OutputInterface $output, string $class): void
+    {
+        $io = new SymfonyStyle($input, $output);
+        $title = sprintf('<info>%s</info>', $class);
+        $rows = [];
+        $dump = new Dumper($output);
+
+        $classMetadata = $this->serializer->getMetadataFor($class);
+
+        foreach ($this->getAttributesData($classMetadata) as $propertyName => $data) {
+            $rows[] = [
+                $propertyName,
+                $dump($data),
+            ];
+        }
+
+        if (!$rows) {
+            if (false === $input->getOption('show-all')) {
+                return;
+            }
+
+            $io->section($title);
+            $io->text('No Serializer data were found for this class.');
+
+            return;
+        }
+
+        $io->section($title);
+
+        $table = new Table($output);
+        $table->setHeaders(['Property', 'Options']);
+        $table->setRows($rows);
+        $table->render();
+    }
+
+    /**
+     * @return array<string, array<string, mixed>>
+     */
+    private function getAttributesData(ClassMetadataInterface $classMetadata): array
+    {
+        $data = [];
+
+        foreach ($classMetadata->getAttributesMetadata() as $attributeMetadata) {
+            $data[$attributeMetadata->getName()] = [
+                'groups' => $attributeMetadata->getGroups(),
+                'maxDepth' => $attributeMetadata->getMaxDepth(),
+                'serializedName' => $attributeMetadata->getSerializedName(),
+                'ignore' => $attributeMetadata->isIgnored(),
+                'normalizationContexts' => $attributeMetadata->getNormalizationContexts(),
+                'denormalizationContexts' => $attributeMetadata->getDenormalizationContexts(),
+            ];
+        }
+
+        return $data;
+    }
+
+    /**
+     * @return string[]
+     */
+    private function getResourcesByPath(string $path): array
```

Comment:
I don't really like this discovery logic, it's fragile and it scans way more files than needed.
I had a look at the code and discovered `param('serializer.mapping.cache.file')`.
Can't we rely on this instead?

Reply:
Yeah Indeed, I think it will be better.

Reply:
I think that's possible, here is an example of the cached file
https://gist.github.com/loic425/9db59617b5c15fe02111ce7df5d3c63b

What do you think? 
we can use the first array keys and replace `_` by `\`. But is it really safe? 

@dunglas, any idea?

Reply:
@nicolas-grekas That makes me think of sth... If we use this file, we cannot provide a path (of a directory) in the command as we did in the debug:validator command and on this command too.
https://github.com/symfony/symfony/blob/9fced6d383b8bfa47983505e94ff18def2dba08b/src/Symfony/Component/Serializer/Command/DebugCommand.php#L49

Reply:
@nicolas-grekas I suggest to remove the path argument and use this debug for one class or for all class that have serializer mapping.

it's easy to parse the serializer.mapping.cache.file
https://3v4l.org/EAquc

Reply:
That does not work cause this file can be empty even if there are some serializer configuration.

```php
<?php

// This file has been auto-generated by the Symfony Cache Component.

return [[


], [


]];

```